base/file.jl

```diff
@@ -356,12 +339,33 @@ is an open file object for this path.
 mktemp(parent)
 
 """
-    mktempdir(parent=tempdir())
+    mktempdir(parent=tempdir(); prefix="$temp_prefix")
 
 Create a temporary directory in the `parent` directory and return its path.
-If `parent` does not exist, throw an error.
+If `parent` does not exist, throw an error. An optional `prefix` to the directory name can
+be provided.
 """
-mktempdir(parent)
+function mktempdir(parent=tempdir(); prefix="$temp_prefix")
+    template = prefix*"XXXXXX"
+    tpath = joinpath(parent, template)
+
+    req = Libc.malloc(_sizeof_uv_fs)
```

Comment:
I guess I now get the approach here. In `jl_uv.c` all request structures are allocated in stack (declared as local var) and passed on to the function in the c-layer. The cleanup is carried out on the address to the request like &req. In the Julia layer you cannot have a stack variable so `zero` is a means to create a Julia UInt8 array which will me memory managed in the Julia layer. Hence, no special action on a throw and clean up will be called on this object after the request object is used in the `ccall`. The approach is a bit hard to understand initially.  Are `try`...`catch` overheads very significant? I can change the code to the `zero` approach.